### PR TITLE
Added User and Login controller in repository pattern.

### PR DESCRIPTION
**Pull request by nidhisonar**
Added User and Login controller in repository pattern.

### CODE REVIEW

File: Maxis/Controllers/LoginController.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        {
+            var functionReturnValue = false;
+            var server = WebConfigurationManager.AppSettings["Ldapserver"];
+            var ldapUser = WebConfigurationManager.AppSettings["Ldapusername"];
+            var ldapPassword = WebConfigurationManager.AppSettings["Ldappassword"];
+           
+            try
+            {
+
+                //using (var pCtx = new PrincipalContext(ContextType.ApplicationDirectory, server, "O=users", ContextOptions.SimpleBind, ldapUser, ldapPassword))
```
Comment (kamalakar1981) [inline Maxis/Controllers/LoginController.cs]: remove commented code

File: Maxis/Controllers/LoginController.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                //{
+                //    functionReturnValue = pCtx.ValidateCredentials(userName, password);
+                //}
+
+                //if (functionReturnValue)
+                //{
+                    var roles = Validate(new LoginViewModel
```
Comment (kamalakar1981) [inline Maxis/Controllers/LoginController.cs]: use meaningful name

**Comment by nidhisonar** [comment]
Added User and Login controller in repository pattern.

File: Maxis/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        public JsonResult Edit(long id)
+        {
+            return Json(_userService.SelectById(id), JsonRequestBehavior.AllowGet);
+        }
+
+        // PUT: USER/EditUser/2
+        [HttpPost]
+
+        public void UpdateUser(EditUserViewModel model)
+        {
```
Comment (kamalakar1981) [inline Maxis/Controllers/UserController.cs]: follow naming convention

File: Maxis/Database/CustomPrincipal.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿
+using System.Security.Principal;
+
+
+namespace Maxis.Database
+{
+    public class CustomPrincipal : ICustomPrincipal
+    {
+        public IIdentity Identity { get; }
```
Comment (kamalakar1981) [inline Maxis/Database/CustomPrincipal.cs]: why this Model class to be there in DB layer?

File: Maxis/Infrastructure/Repositories/MapRepository.cs [hunk 5]
```diff
@@ -77,10 +77,10 @@ public List<CableViewModel> getCableDetails(string cableType)
             return db.ONNET_SRCH_OSP_CABLE.Where(m => m.CABLE_TYPE.ToUpper() == cableType.ToUpper()).ProjectTo<CableViewModel>().ToList();
         }
 
-        public List<BuildingViewModel> getBuildingDetails(string NEName)
+        public List<BuildingViewModel> getBuildingDetails(DbGeography searchPoint, int range)
         {
             db.Configuration.ProxyCreationEnabled = false;
-            return db.ONNET_SRCH_NE_ABE.Where(m => m.NE_NAME.ToUpper() == NEName.ToUpper()).ProjectTo<BuildingViewModel>().ToList();
+            return db.ONNET_SRCH_NE_ABE.Where(m => m.NE_NAME.ToUpper() == " ").ProjectTo<BuildingViewModel>().ToList();
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/MapRepository.cs]: use string.Empty instead

File: Maxis/Infrastructure/Repositories/UserRepository.cs [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                            Email = null,
+                            Mobile = null,
+                            Department = null,
+                            Title = null,
+                            Status = null,
+                            RoleId = 3
+                        };
+
+                        using (var context = new MaxisEntities())
+                        {
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: why new context is required, can't we use same _db ?

File: Maxis/Infrastructure/Repositories/UserRepository.cs [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+        public void Update(EditUserViewModel model)
+        {
+           
+            try
+            {
+                //using (var entities = new MaxisEntities())
+                //{
+                    var entity = _db.ONNET_USER.FirstOrDefault(u => u.UserId == model.UserId);
+                    if (entity != null)
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: check for entity.Any() instead

**Pull request merged**